students/MzKhan/lesson09/donor_models.py

```diff
@@ -0,0 +1,158 @@
+'''
+Name: Muhammad Khan
+Date: 04/09/2019
+Assignment09
+'''
+
+import os
+import re
+import datetime
+from operator import itemgetter as igetter
+
+class Donor:
+    # The donor class manages the functionality for a single donor.
+
+    def __init__(self, name, donations):
+        #Initializer or Constructor
+        # Make sure that the data is added to a list.
+        self._name = name
+        if isinstance(donations, list):
+            self._donations = donations
+        elif isinstance(donations, tuple):
+            self._donations = list(donations)
+        else:
+            self._donations = [donations]
+
+
+    def __repr__(self):
+        """Reproduce the Donor Object"""
+        return "Donor({},{})".format(self._name, self._donations)
+
+
+    def add_donation(self, amount):
+        """Add the the donation amount for a donor"""
+        self._donations.append(amount)
+
+    #Properties for attribute-like access.
+    @property
+    def name(self):
+        return self._name
+
+
+    @property
+    def donations(self):
+        return self._donations
+
+
+    @property
+    def total_donations(self):
+        return len(self._donations)
+
+
+    @property
+    def total_donations_amount(self):
+        return sum(self._donations)
+
+
+    @property
+    def avg_donation(self):
+        return self.total_donations_amount / self.total_donations
+
+
+    def thank_you(self, name, amount):
+        """
+        :parm 1: name - required argument
+        :parm 2: amount - requried argument.
+        :return: string - a formatted email message for the donor.
+        """
+        email_msg = """
+        \rDear {:},
+
+        \rThank you so much for your generous donation of $ {:.2f}.
+
+        \rBest Regards,
+
+        \r-Team"""
+        return email_msg.format(name, amount)
+
+
+
+class DonorCollection:
+    # This class contains the functionality for all the donors in the donors'
+    # database.
+
+    def __init__(self, *args):
+        #dict comprehension
+        self.donors = {d.name: d for d in args}
+
+
+    def add_donation(self, name, donation):
+        if self.donors.get(name):
+            self.donors[name].add_donation(donation)
+        else:
+            self.donors[name] = Donor(name, donation)
+
+
+    @property
+    def list(self):
+        """Return the key values ( donor names ) from the dict."""
+        return list(self.donors)
+
+
+    @staticmethod
+    def sort_desc(a_list):
+        return sorted(a_list, key=igetter(1), reverse=True)
```

Comment:
isn't `a_list` data thats part of your instance? `return sorted(self.donors, key=igetter(1), reverse=True)`